queue.go

```diff
@@ -31,3 +31,17 @@ func (queue *Queue) delete(receiptHandle string) (didDelete bool) {
 
 	return
 }
+
+func (queue *Queue) Messages() (*orderedmap.OrderedMap, bool) {
```

Comment:
This is a totally reasonable approach, with one modification:

`Messages` should ideally block write access to the queue. This will gaurantee under a concurrent environment that a true snapshot is taken, but probably more importantly you don't need to check `Get` since all keys will exist while write access is blocked. However, to do this you would need to create a `sync.RWMutex` on the `Queue`, then add the correct locks to each method (not just `Messages`).

This also has the added bonus that orderedmap itself is not thread-safe so the current implementation of mocksqs could actually mangle the queue:

```go
type Queue struct {
 	sync.RWMutex
 	// ...
}

func (queue *Queue) Messages() *orderedmap.OrderedMap {
 	queue.Lock()
 	defer queue.Unlock()

 	rv := orderedmap.NewOrderedMap()
 	for el := m.Front(); el != nil; el = el.Next() {
 		rv.Set(el.Key, el.Value)
 	}

 	return rv
}
```

For bonus points, you could create a PR for orderedmap to provide a `Copy` method.

Reply:
Thanks. That is a very clear and helpful response 🙏 😄 

I've decided to go for the bonus points and add a `Copy()` function (see [PR #19](https://github.com/elliotchance/orderedmap/pull/19)).